src/library-authoring/LibraryLayout.tsx

```diff
@@ -27,28 +27,31 @@ const LibraryLayout = () => {
 
   return (
     <LibraryProvider
+      /** We need to pass the collectionId as key to the LibraryProvider to force a re-render
+      * when we navigate to a collection page. */
       key={collectionId}
       libraryId={libraryId}
       collectionId={collectionId}
-      /** The component picker modal to use. We need to pass it as a reference instead of
-       * directly importing it to avoid the import cycle:
-       * ComponentPicker > LibraryAuthoringPage/LibraryCollectionPage >
-       * Sidebar > AddContentContainer > ComponentPicker */
+        /** The component picker modal to use. We need to pass it as a reference instead of
+         * directly importing it to avoid the import cycle:
+         * ComponentPicker > LibraryAuthoringPage/LibraryCollectionPage >
+         * Sidebar > AddContentContainer > ComponentPicker */
       componentPicker={ComponentPicker}
     >
-      <Routes>
-        <Route
-          path="collection/:collectionId"
-          element={<LibraryCollectionPage />}
-        />
-        <Route
-          path="*"
-          element={<LibraryAuthoringPage />}
-        />
-      </Routes>
-      <CreateCollectionModal />
-      <ComponentEditorModal />
-      <LibraryTeamModal />
```

Comment:
Once the `LibraryTeamModal` was used only inside the `LibraryInfo`, I moved the Modal there and removed the methods from the `LibraryContext`.